core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -400,6 +396,11 @@ private[pickling]  trait SourceGenerator extends Macro with tags.FastTypeTagMacr
         implicit object $picklerName extends _root_.scala.pickling.Pickler[$tpe] with _root_.scala.pickling.Generated {
           def pickle(picklee: $tpe, builder: _root_.scala.pickling.PBuilder): _root_.scala.Unit = ${genPicklerLogic[T](picklerAst)}
           val tag: _root_.scala.pickling.FastTypeTag[$tpe] = $createTagTree
+
+          if(_root_.scala.pickling.internal.currentRuntime.picklers.lookupPickler(tag.key).isEmpty) {
```

Comment:
So I hate to be a negative nancy, but I think we should actually be more aggressive here.  We should not even instantiate our class if there is a registered pickler for our type.

As an interim though, this should be ok :).

Also, you should look into (possibly) ALWAYS creating pickler/unpickler, because statically creatinga  pickler and trying to runtime-instantiate an unpickler actually has subtle bugs.  If we can do one side, we should do both sides and register them both....